### DISCUSSION TITLE
Started seeing "complete is not a known entry point.." in new terms with kitty 0.27.0?

### DISCUSSION THREAD

**Discussion by khughitt** [kovidgoyal/kitty#5972] [Q&A] [answered]
Greetings!

After updating kitty on Arch Linux to 0.27.0, I started getting a kind of message/warning each time I launch kitty:

> complete is not a known entry point. Choices are: icat, list-fonts, hold, runpy, launch, open, kitten, edit-config, shebang, edit

The only place in my config where "complete" is mentioned is in a font setting, so I thought perhaps that was somehow not being parsed correctly:

> font_family      Fura Code Regular Nerd Font Complete Mono

However, commenting out the relevant font entries doesn't make a difference..

Any ideas what the issue could be?

Full config:

```
font_family      Fura Code Regular Nerd Font Complete Mono
bold_font        Fira Code Bold Nerd Font Complete
italic_font      Source Code Pro Italic
bold_italic_font Source Code Pro Bold Italic

font_features FiraCode-Retina +ss02 +ss03 +ss04 +ss05 +zero

scrollback_lines 8192
enable_audio_bell no

background_opacity 1.0
window_padding_width 4

default_pointer_shape arrow

cursor_stop_blinking_after 0.0

strip_trailing_spaces smart

#: Color scheme
#include color.conf
include ${HOME}/.cache/wal/colors-kitty.conf

# overide some colors
foreground               #abb2bf
background               #282c34
selection_foreground     #5c6370
selection_background     #abb2bf
cursor                   #abb2bf
cursor_text_color        #5c6370

#: Host-specific options
include ${HOSTNAME}.conf
```

```
# host-specific config
font_size 15.0
```

```
# pywal
foreground   #c5c5c5
background   #171717
cursor       #c5c5c5

color0       #171717
color8       #515151
color1       #484951
color9       #484951
color2       #5d5151
color10      #5d5151
color3       #7a6f68
color11      #7a6f68
color4       #8695a2
color12      #8695a2
color5       #c69671
color13      #c69671
color6       #9fb1be
color14      #9fb1be
color7       #c5c5c5
color15      #c5c5c5
```

**Comment by kovidgoyal** [comment, marked as answer]
You have an obsolete invocation to kitty complate in your shell rc file. Remove it.

**Reply by page-down** [reply]
@kovidgoyal
The completion script carried by fish will be present in some Linux distributions for some time.
https://github.com/fish-shell/fish-shell/blob/3.6.0/share/completions/kitty.fish
```fish
... kitty +complete fish2
```

The issue will likely occur when using the latest fish and the latest kitty when shell integration is disabled (e.g. using ssh to connect to, rather than ssh kitten).

Is there any chance to keep `kitty +complete` in recent versions (such as delegating to kitten) to keep backward compatibility and smooth transition to new versions? Thanks.

**Reply by khughitt** [reply]
Thanks! I didn't even think to check there.. Thanks for all of your work on Kitty, too! It's much appreciated. :pray:

**Comment by kovidgoyal** [comment]
I am fine with it, feel free to send a PR for it. But note that it will
require a fair bit of code to actually implement completion in the old
style. You can instead just have it return nothing so the user isnt
bothered by an error message. Completion not actually working is not
that big of a deal in this rare scenario.

**Reply by page-down** [reply]
Currently there is also `kitty +complete setup` being used.
This also needs to be rewritten in Go.

https://github.com/archlinux/svntogit-community/blob/835f27dfcb1b84794548319df46cf7d513715dfb/trunk/PKGBUILD#L38-L41

https://github.com/NixOS/nixpkgs/blob/3f140e028d3778c940fb31a233ca349ec4dd6eb4/pkgs/applications/terminal-emulators/kitty/default.nix#L184-L187

I think the completion script submitted to fish can be output directly by `setup`, so that future changes will take effect immediately.

fish-shell/share/completions/kitty.fish
```fish
kitten __complete__ setup fish kitty | source -
```

fish-shell/share/completions/kitten.fish
```fish
kitten __complete__ setup fish kitten | source -
```
